### DISCUSSION TITLE
Custom transaction are not sent?

### DISCUSSION THREAD

**Discussion by BarbecueSilver** [getsentry/sentry-react-native#2588] [Q&A]
Hej there!

Thanks for your project and the effort.

I just encountered, that following your documentation ([Performance](https://docs.sentry.io/platforms/react-native/performance/), [Custom Transaction](https://docs.sentry.io/platforms/react-native/performance/instrumentation/custom-instrumentation/)) will not lead to a sent Transaction at all.

It is stated in the *Custom Instrumentation* part as first example:
```Typescript
const transaction = Sentry.startTransaction({ name: "test-transaction" });
const span = transaction.startChild({ op: "functionX" }); // This function returns a Span
// functionCallX
span.finish(); // Remember that only finished spans will be sent with the transaction
transaction.finish(); // Finishing the transaction will send it to Sentry
```

We do not receive any Transaction to our Sentry project with this code. No worries, I set `tracesSampleRate` to `1` for testing purposes 👍 
I just received this in our console:
```console
 LOG  Sentry Logger [log]: [Tracing] starting undefined transaction - test-transaction
 LOG  Sentry Logger [log]: [Tracing] Starting 'functionX' span on transaction 'test-transaction' (99dfe0f5f58d782a).
 LOG  Sentry Logger [log]: [Tracing] Finishing 'functionX' span on transaction 'test-transaction' (99dfe0f5f58d782a).
 LOG  Sentry Logger [log]: [Tracing] Finishing undefined transaction: test-transaction.
```

I dig around and found that maybe the property `op` was missing in `startTransaction`.
Indeed! After adding it, we finally received the Transaction in our Sentry project 🎉 
```Typescript
const transaction = Sentry.startTransaction({ name: "test-transaction", op: "wohooo-I-am-working-now" });
const span = transaction.startChild({ op: "functionX" }); // This function returns a Span
// functionCallX
span.finish(); // Remember that only finished spans will be sent with the transaction
transaction.finish(); // Finishing the transaction will send it to Sentry
```

---

Now I am asking myself:

Is it sent but not shown in the project?
Is this is a bug? (because `op` is `undefined`; a bug on the sentry page or in ReactNative package)
Missing documentation? (`op` might be mandatory on `transaction`)
Shall I open an issue for it? (I will! just let me know 👍 )

**Comment by bukhari-flip** [comment]
The same thing happened when I try to put a custom timestamp on a transaction and/or span

```js
const transactionStartTimeStamp = (new Date()).getTime();
// functionCallX
const spanStartTimeStamp = (new Date()).getTime();
// functionCallX-Y
const spanEndTimeStamp = (new Date()).getTime();
const transactionEndTimeStamp = (new Date()).getTime();

const transaction = Sentry.startTransaction({
  name: "test-transaction",
  op: "functionX",
  startTimestamp: transactionStartTimeStamp,
});
const span = transaction.startChild({
  op: "functionX-Y",
  startTimestamp: spanStartTimeStamp,
});
span.finish(spanEndTimeStamp);
transaction.finish(transactionEndTimeStamp);
```

When the log activates, it will log both the start & finishing event.
The `startTimeStamp` when starting event/span was found from the SpanContext.
```js
(property) SpanContext.startTimestamp?: number | undefined
/// Timestamp in seconds (epoch time) indicating when the span started.
```

And so the param when finish called.
```js
(method) Span.finish(endTimestamp?: number | undefined): void
/// Sets the finish timestamp on the current span.
/// 
/// @param endTimestamp — Takes an endTimestamp if the end should not be the time when you call this function.
```

**Reply by bukhari-flip** [reply]
My bad, I used to divide time by 1000, since `getTime()` returns milliseconds.
